extensions/ql-vscode/src/query-history.ts

```diff
@@ -191,12 +192,26 @@ export class QueryHistoryManager {
 
   async handleShowQueryLog(queryHistoryItem: CompletedQuery) {
     if (queryHistoryItem.logFileLocation) {
+      const uri = vscode.Uri.parse(queryHistoryItem.logFileLocation);
       try {
-        await vscode.window.showTextDocument(vscode.Uri.parse(queryHistoryItem.logFileLocation), {
-          viewColumn: vscode.ViewColumn.Beside
+        await vscode.window.showTextDocument(uri, {
         });
       } catch (e) {
-        helpers.showAndLogErrorMessage(`Could not open log file ${queryHistoryItem.logFileLocation}`);
+        if (e.message.includes('Files above 50MB cannot be synchronized with extensions')) {
+          const res = await helpers.showBinaryChoiceDialog('File is too large to open in the editor, do you want to open exterally?');
+          if (res) {
+            try {
+              await vscode.commands.executeCommand('revealFileInOS', uri);
+            } catch (e) {
+              logger.log(e.message);
```

Comment:
Maybe a `showAndLogErrorMessage` here, too, so the user isn't surprised by silence when the `revealFileInOS` fails.

Reply:
Will do.